src/amo/components/SearchSort/SearchSort.scss

```diff
@@ -65,6 +73,10 @@
     background: transparentize($link-color, 0.8);
     box-shadow: none;
   }
+
+  @include respond-to(large) {
```

Comment:
Any time we're using `display: none` we should also consider the alternative of with-holding the content. 

Of changing content output based on UA or the mamo cookie is not equal to a viewport size based decision. See also https://github.com/mozilla/addons-frontend/issues/2355 (this is probably something for later rather than now).